src/utils/neynar/openframes/neynarValidator.ts

```diff
@@ -0,0 +1,54 @@
+import { OpenFramesRequest, RequestValidator, ValidationResponse } from '@open-frames/types';
+import { FarcasterOpenFramesRequest, FrameRequest, FrameValidationData } from '../../../core/types';
+import { NEYNAR_DEFAULT_API_KEY, neynarFrameValidation } from '../frame/neynarFrameFunctions';
+import { Options } from './types';
+
+export class NeynarValidator
+  implements RequestValidator<FarcasterOpenFramesRequest, FrameValidationData, 'farcaster'>
+{
+  readonly protocolIdentifier = 'farcaster';
+  private options: Required<Options>;
+
+  constructor(options?: Options) {
+    this.options = {
+      neynarApiKey: options?.neynarApiKey || NEYNAR_DEFAULT_API_KEY,
+      castReactionContext: options?.castReactionContext || true,
+      followContext: options?.followContext || true,
+    };
+  }
+
+  minProtocolVersion(): string {
+    return `${this.protocolIdentifier}@VNext`;
+  }
+
+  isSupported(payload: OpenFramesRequest): payload is FarcasterOpenFramesRequest {
+    const isCorrectClientProtocol =
+      !!payload.clientProtocol && payload.clientProtocol.startsWith('farcaster@');
+    const isTrustedDataValid = typeof payload.trustedData?.messageBytes === 'string';
+
+    return isCorrectClientProtocol && isTrustedDataValid;
+  }
```

Comment:
So I see, `isSupported` is where the magic happens.

And ideally we can have as many client as we want.

There are some implementation details that we need to align, but I like where your head is @neekolas

Reply:
Exactly. [Here is an example of the XMTP RequestValidator](https://github.com/xmtp/xmtp-node-js-tools/pull/147) that can be safely passed in to `getOpenFramesMessage` as an argument and will allow for processing of button clicks coming from XMTP.